includes/class-llms-block-templates.php

```diff
@@ -0,0 +1,564 @@
+			'archive-llms_membership'    => esc_html__( 'Membership Catalog', 'lifterlms' ),
+			'single-certificate'         => esc_html__( 'Single Certificate', 'lifterlms' ),
+			'single-no-access'           => esc_html__( 'Single Requiring Membership', 'lifterlms' ),
+			'taxonomy-course_cat'        => esc_html__( 'Taxonomy Course Category', 'lifterlms' ),
+			'taxonomy-course_difficulty' => esc_html__( 'Taxonomy Course Difficulty', 'lifterlms' ),
+			'taxonomy-course_tag'        => esc_html__( 'Taxonomy Course Tag', 'lifterlms' ),
+			'taxonomy-course_track'      => esc_html__( 'Taxonomy Course Track', 'lifterlms' ),
+			'taxonomy-memberhsip_cat'    => esc_html__( 'Taxonomy Membership Category', 'lifterlms' ),
+			'taxonomy-memberhsip_tag'    => esc_html__( 'Taxonomy Membership Tag', 'lifterlms' ),
+		);
+
+		$this->template_titles = $this->template_titles ?? $template_titles;
+
+		/**
+		 * Filters the block template titles.
+		 *
+		 * @since [version]
+		 *
+		 * @param array $template_titles  {
+		 *     Associative array of template titles.
+		 *
+		 *     @type string $slug  The template slug (unprefixed).
+		 *     @type string $title The template readable titles.
+		 * }
+		 */
+		return apply_filters( 'lifterlms_block_templates_titles', $this->template_titles );
+
+	}
+
+	/**
+	 * Block Templates admin js strings.
+	 *
+	 * @since [version]
+	 *
+	 * @param string[] $strings Localization admin strings.
+	 * @return string[]
+	 */
+	public function add_block_templates_admin_strings_js( $strings ) {
+
+		return array_merge(
+			$strings,
+			array(
+				'LifterLMS Course Catalog Template'               => esc_html__( 'LifterLMS Course Catalog Template', 'lifterlms' ),
+				'LifterLMS Membership Catalog Template'           => esc_html__( 'LifterLMS Membership Catalog Template', 'lifterlms' ),
+				'LifterLMS Single Template requiring Membership'  => esc_html__( 'LifterLMS Single Template requiring Membership', 'lifterlms' ),
+				'LifterLMS Certificate Template'                  => esc_html__( 'LifterLMS Certificate Template', 'lifterlms' ),
+				'LifterLMS Course Category Taxonomy Template'     => esc_html__( 'LifterLMS Course Category Taxonomy Template', 'lifterlms' ),
+				'LifterLMS Course Difficulty Taxonomy Template'   => esc_html__( 'LifterLMS Course Difficulty Taxonomy Template', 'lifterlms' ),
+				'LifterLMS Course Tag Taxonomy Template'          => esc_html__( 'LifterLMS Course Tag Taxonomy Template', 'lifterlms' ),
+				'LifterLMS Course Track Taxonomy Template'        => esc_html__( 'LifterLMS Course Track Taxonomy Template', 'lifterlms' ),
+				'LifterLMS Membership Category Taxonomy Template' => esc_html__( 'LifterLMS Membership Category Taxonomy Template', 'lifterlms' ),
+				'LifterLMS Membership Tag Taxonomy Template'      => esc_html__( 'LifterLMS Membership Tag Taxonomy Template', 'lifterlms' ),
+			)
+		);
+
+	}
+
+}
+
+$block_templates = new LLMS_Block_Templates();
+$block_templates->init();
+
+return $block_templates;
```

Comment:
The purpose of this is that I want to be able to call this object methods from another file without adding the hooks.
See: 
https://github.com/gocodebox/lifterlms/pull/1952/files#diff-7947d18155c2e03348f57a9ef19a46d9e47a3a999f09f521726fb3c2e82a7be7R466

that `init()` should be actually be renamed `hooks()` or something of the sort.
Hooks addition are not `initialization` to me.

If you prefer having them in the constructor I'd say that this must become a singleton, and I add a `block_templates()` method here:
https://github.com/gocodebox/lifterlms/blob/dev/class-lifterlms.php

what do you think?

Reply:
Mostly what I don't like here is having functionality / logic buried at the bottom of a class file, especially since we *don't do this anywhere else* it makes it hard to locate in the future since we're not primed to look for any functional code at the bottom of a class file.

I think, based on existing patterns, the singleton makes the most sense then.

Reply:
@thomasplevy singleton thing done